File: FrontEnd/src/types/payment.ts
export type Payment = {
  id: string;
  tenantId: string;
  invoiceId: string;
  customerId: string;
  invoiceNumber: string;
  customerName: string;
  customerCompanyName: string;
  amount: number;
  currency: string;
  paymentDate: string;
  referenceNumber: string;
  notes: string;
  createdAtUtc: string;
};

export type CreatePaymentRequest = {
  invoiceId: string;
  amount: number;
  currency: string;
  paymentDate: string;
  referenceNumber: string;
  notes: string;
};

File: BackEnd/src/CollectFlow.Domain/Entities/Payment.cs
namespace CollectFlow.Domain.Entities;

public class Payment : BaseEntity
{
    public Guid TenantId { get; set; }

    public Guid InvoiceId { get; set; }
    public Invoice Invoice { get; set; } = default!;

    public Guid CustomerId { get; set; }
    public Customer Customer { get; set; } = default!;

    public decimal Amount { get; set; }
    public string Currency { get; set; } = "USD";
    public DateOnly PaymentDate { get; set; }
    public string ReferenceNumber { get; set; } = string.Empty;
    public string Notes { get; set; } = string.Empty;
}